src/test/java/com/jwplayer/southpaw/util/ByteArraySetTest.java

```diff
@@ -46,6 +74,651 @@ public ByteArraySet createSmallSet() {
+
+        List<ByteArray> vals = getRandomByteArrays(size);
+
+        for (ByteArray val: vals) {
+            assertTrue(set.add(val));
+        }
+        assertEquals(size, set.size());
+
+        List<ByteArray> stagedVals = getRandomByteArrays(size, getDifferentRandomSeed());
+        Set<ByteArray> extraVals;
+        if (forceByteArraySet) {
+            extraVals = new ByteArraySet();
+        } else {
+            extraVals = new HashSet<ByteArray>();
+        }
+        while(extraVals.size() < size) {
+            if (stagedVals.size() == 0) {
+                stagedVals = getRandomByteArrays(size, getDifferentRandomSeed());
+            }
+            ByteArray val = stagedVals.get(0);
+            stagedVals.remove(0);
+            if (!vals.contains(val) && !extraVals.contains(val)) {
+                extraVals.add(val);
+            }
+        }
+
+        set.addAll(extraVals);
+        assertEquals(2 * size, set.size());
+
+        for (ByteArray val: vals) {
+            assertTrue(set.contains(val));
+        }
+        for (ByteArray val: extraVals) {
+            assertTrue(set.contains(val));
+        }
+        for (ByteArray val: set) {
+            assertTrue(vals.contains(val) || extraVals.contains(val));
+        }
+    }
+
+    @Test
+    public void emptyAddAll() {
+        testAddAll(0, false);
+        testAddAll(0, true);
+    }
+
+    @Test
+    public void reallySmallAddAll() {
+        testAddAll(1, false);
+        testAddAll(1, true);
+    }
+
+    @Test
+    public void smallAddAll() {
+        testAddAll(2, false);
+        testAddAll(2, true);
+    }
+
+    @Test
+    public void regularAddAll() {
+        testAddAll(SIZE_REGULAR, false);
+        testAddAll(SIZE_REGULAR, true);
+    }
+
+    @Test
+    public void bigAddAll() {
+        testAddAll(SIZE_BIG, false);
+        testAddAll(SIZE_BIG, true);
+    }
+
+    @Test
+    public void reallyBigAddAll() {
+        testAddAll(SIZE_REALLY_BIG, false);
+        testAddAll(SIZE_REALLY_BIG, true);
+    }
+
+    private void testSimilarAdd(boolean forceMerger) {
+        System.out.println("Testing add method for similar inputs with " + forceMerger + " forceMerger");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = new ArrayList<>();
+        vals.add(new ByteArray("bytear"));
+        vals.add(new ByteArray("ByteAr"));
+        vals.add(new ByteArray("bytea0"));
+        vals.add(new ByteArray("0ytear"));
+
+        List<ByteArray> insertedVals = new ArrayList<>();
+        for (ByteArray val: vals) {
+            if (forceMerger) {
+                set.serialize();
+            }
+            assertTrue(set.add(val));
+            assertTrue(set.contains(val));
+            insertedVals.add(val);
+            assertEquals(insertedVals.size(), set.size());
+        }
+
+        assertEquals(4, set.size());
+
+        for (ByteArray val: vals) {
+            assertTrue(set.contains(val));
```

Comment:
This looks like a duplicate of the above assertion on line 248. Am I missing something?